core/src/main/scala/kafka/server/DelayedFetch.scala

```diff
@@ -139,6 +140,9 @@ class DelayedFetch(delayMs: Long,
           case _: NotLeaderForPartitionException =>  // Case A
             debug("Broker is no longer the leader of %s, satisfy %s immediately".format(topicPartition, fetchMetadata))
             return forceComplete()
+          case _: ReplicaNotAvailableException =>  // Case H
+            debug("Broker no longer has a replica of %s, satisfy %s immediately".format(topicPartition, fetchMetadata))
```

Comment:
Can we use string interpolation? We do it for 3 of the other 3 cases, but not the one immediately above. Also, should this be case `B` and shift the rest instead of case `H`?  It seems like it would be easier to understand if the first 3 cases were about leader/replica/partition knowledge. Also, we could have the case checks in order: A, B, C, etc.

What do you think?

Reply:
@ijuma Thanks for the review, updated.